tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,563 @@
+	}
+
+	if (chan->chan_id == 0) {
+		/*By the time we rx on chan 0, we should have alrady received on chan1*/
+		if ((channels[1].sdus_received != (channels[0].sdus_received + 1))) {
+			FAIL("Didn't receive on channel 1 first: channels[0].sdus_received:%i channels[1].sdus_received:%i",
+			     channels[0].sdus_received,
+			     channels[1].sdus_received);
+		}
+	}
+
+	chan->sdus_received++;
+
+	return 0;
+}
+
+static void chan_sent_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	chan->buf = 0;
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static volatile int num_connect_chans;
+static void chan_connected_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+
+	LOG_DBG("tx.mtu %d, tx.mps: %d, rx.mtu: %d, rx.mps %d", sys_cpu_to_le16(chan->le.tx.mtu),
+		sys_cpu_to_le16(chan->le.tx.mps), sys_cpu_to_le16(chan->le.rx.mtu),
+		sys_cpu_to_le16(chan->le.rx.mps));
+
+	num_connect_chans++;
+}
+
+static void chan_disconnected_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+
+	chan->in_use = false;
+	num_connect_chans--;
+}
+
+static void chan_status_cb(struct bt_l2cap_chan *l2cap_chan, atomic_t *status)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d, status: %ld", chan->chan_id, *status);
+}
+
+static void chan_released_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static void chan_reconfigured_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static const struct bt_l2cap_chan_ops l2cap_ops = {
+	.alloc_buf = chan_alloc_buf_cb,
+	.recv = chan_recv_cb,
+	.sent = chan_sent_cb,
+	.connected = chan_connected_cb,
+	.disconnected = chan_disconnected_cb,
+	.status = chan_status_cb,
+	.released = chan_released_cb,
+	.reconfigured = chan_reconfigured_cb,
+};
+
+static void connect_num_channels(uint8_t num_l2cap_channels)
+{
+	struct channel *chan = NULL;
+
+	#define L2CAP_ECRED_CHAN_MAX 5
+	struct bt_l2cap_chan *allocated_channels[L2CAP_ECRED_CHAN_MAX] = { NULL };
+	#undef L2CAP_ECRED_CHAN_MAX
+	int err = 0;
+
+	for (int i = 0; i < num_l2cap_channels; i++) {
+		chan = get_free_channel();
+		if (!chan) {
+			FAIL("failed, chan not free");
+			return;
+		}
+		chan->le.chan.ops = &l2cap_ops;
+		chan->le.rx.mtu = DATA_MTU;
+		chan->le.rx.mps = DATA_MPS;
+		allocated_channels[i] = &chan->le.chan;
+	}
+
+	err = bt_l2cap_ecred_chan_connect(default_conn, allocated_channels, servers[0].psm);
+	if (err) {
+		FAIL("can't connect ecred %d ", err);
+	}
+}
+
+static void disconnect_all_channels(void)
+{
+	int err = 0;
```

Comment:
Wrong location and unnecessary initialisation upon declaration. Variables should always be declared in the smallest possible scope, so in this case it should go inside the if-branch.